data_preparation/global_AgERA5/download_agera5.py

```diff
@@ -0,0 +1,123 @@
+import os
+import cdsapi
+import concurrent.futures
+
+# Most of the code here comes from Abdelrahman's library.
+# https://github.com/AbdelrahmanAmr3/earthstat
+
+AgERA5_params = {
+    "Maximum_Temperature": ("2m_temperature", "24_hour_maximum"),
+    "Minimum_Temperature": ("2m_temperature", "24_hour_minimum"),
+    "Mean_Temperature": ("2m_temperature", "24_hour_mean"),
+    "Solar_Radiation_Flux": ("solar_radiation_flux", None),
+    "Precipitation_Flux": ("precipitation_flux", None),
+}
+
+
+# @author: Abdelrahman Saleh
+def get_agera5_params(sel_param, start_year, end_year, area=None):
```

Comment:
The API of CDS allows users to download for a maximum of one year; therefore, this limitation is passed by multiprocessing to send multiple requests to the API server.

So, in this function, replacing start_year and end_year with just one variable year makes more sense, and keep the start_year and end_year just in multiprocessing function.